PhysicsTools/PatAlgos/plugins/PATLostTracks.cc

```diff
@@ -67,85 +108,123 @@ pat::PATLostTracks::~PATLostTracks() {}
+	    if(trkStatus[trkIt->key()]==TrkStatus::NOTUSED)  trkStatus[trkIt->key()]=TrkStatus::VTX;
 	}
     }
 
-    std::vector<int> mapping(tracks->size(),-1);
-    int j=0;
-    for(unsigned int i=0; i < used.size(); i++)
-    {
-	const reco::Track & tr = (*tracks)[i];
-	if(used[i] == 2 || 
-	  (used[i]==0 && tr.pt() > minPt_ && tr.numberOfValidHits() >= minHits_ && tr.hitPattern().numberOfValidPixelHits() >= minPixelHits_ ) 
-          )
-		{
-			outPtrP->push_back(tr);
-			reco::Candidate::PolarLorentzVector p4(tr.pt(),tr.eta(),tr.phi(),0.13957018);
-			outPtrC->push_back(pat::PackedCandidate(p4,tr.vertex(),tr.phi(),211*tr.charge(),PVRefProd,PV.key()));
-			outPtrC->back().setTrackProperties((*tracks)[i]);
-		        if(PVOrig.trackWeight(edm::Ref<reco::TrackCollection>(tracks,i)) > 0.5) {
-		                outPtrC->back().setAssociationQuality(pat::PackedCandidate::UsedInFitTight);
-			}
-
-			mapping[i]=j;
-			j++;
-		}
+    std::vector<int> mapping(tracks->size(),-1);  
+    int lostTrkIndx=0;
+    for(unsigned int trkIndx=0; trkIndx < tracks->size(); trkIndx++){
+        reco::TrackRef trk(tracks,trkIndx);
+	if( trkStatus[trkIndx] == TrkStatus::VTX || 
+	   (trkStatus[trkIndx]==TrkStatus::NOTUSED && passTrkCuts(*trk)) ) { 
+	
+	    outPtrTrks->emplace_back(*trk);
+	    addPackedCandidate(*outPtrTrksAsCands,trk,pv,pvRefProd,pvOrig,trkStatus[trkIndx]);
+	
+	    //for creating the reco::Track -> pat::PackedCandidate map
+	    //not done for the lostTrack:eleTracks collection
+	    mapping[trkIndx]=lostTrkIndx;
+	    lostTrkIndx++;
+	}else if( (trkStatus[trkIndx]==TrkStatus::PFELECTRON || trkStatus[trkIndx]==TrkStatus::PFPOSITRON ) 
+		  && passTrkCuts(*trk) ) {
+	  addPackedCandidate(*outPtrEleTrksAsCands,trk,pv,pvRefProd,pvOrig,trkStatus[trkIndx]);
+			      
+	
+      }	      
     } 
-    iEvent.put(outPtrP);
-    edm::OrphanHandle<pat::PackedCandidateCollection> oh =   iEvent.put(outPtrC);
-    std::auto_ptr<edm::Association<pat::PackedCandidateCollection> > tk2pc(new edm::Association<pat::PackedCandidateCollection>(oh   ));
+    
+    iEvent.put(std::move(outPtrTrks));
+    iEvent.put(std::move(outPtrEleTrksAsCands),"eleTracks");
+    edm::OrphanHandle<pat::PackedCandidateCollection> oh = iEvent.put(std::move(outPtrTrksAsCands));
+    auto tk2pc = std::make_unique<edm::Association<pat::PackedCandidateCollection>>(oh);
     edm::Association<pat::PackedCandidateCollection>::Filler tk2pcFiller(*tk2pc);
     tk2pcFiller.insert(tracks, mapping.begin(), mapping.end());
     tk2pcFiller.fill() ; 
-    iEvent.put(tk2pc);
+    iEvent.put(std::move(tk2pc));
+
+}  
 
+bool pat::PATLostTracks::passTrkCuts(const reco::Track& tr)const
+{
+  
+    bool passTrkPt = tr.pt() > minPt_;
+    bool passTrkHits = tr.numberOfValidHits() >= minHits_ && 
+                       tr.hitPattern().numberOfValidPixelHits() >= minPixelHits_;
+    bool passTrkQual = passesQuality(tr,qualsToAutoAccept_);
+    return (passTrkPt && passTrkHits) || passTrkQual;
```

Comment:
having this method implementation more aligned with the 91X version would be nice.
cf 
```diff
+    const bool passTrkHits = tr.pt() > minPt_ && 
+                             tr.numberOfValidHits() >= minHits_ && 
+                             tr.hitPattern().numberOfValidPixelHits() >= minPixelHits_;
+    const bool passTrkQual = passesQuality(tr,qualsToAutoAccept_);
+
+    return passTrkHits || passTrkQual;
```